Tests/AdyenTests/Adyen Tests/Components/Affirm/AffirmComponentTests.swift

```diff
@@ -0,0 +1,210 @@
+        sut.delegate = delegate
+        let expectedBillingAddress = PostalAddress(city: "New York",
+                                                   country: "US",
+                                                   houseNumberOrName: "14",
+                                                   postalCode: "10019",
+                                                   stateOrProvince: "AL",
+                                                   street: "8th Ave",
+                                                   apartment: nil)
+        let expectedDeliveryAddress = PostalAddress(city: "Los Angeles",
+                                                    country: "US",
+                                                    houseNumberOrName: "3310",
+                                                    postalCode: "90040",
+                                                    stateOrProvince: "AL",
+                                                    street: "Garfield Ave",
+                                                    apartment: nil)
+        
+        // Then
+        let didSubmitExpectation = expectation(description: "PaymentComponentDelegate must be called when submit button is clicked.")
+        delegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === sut)
+            let details = data.paymentMethod as! AffirmDetails
+            XCTAssertEqual(details.shopperName?.firstName, "Katrina")
+            XCTAssertEqual(details.shopperName?.lastName, "Del Mar")
+            XCTAssertEqual(details.telephoneNumber, "2025550146")
+            XCTAssertEqual(details.emailAddress, "katrina@mail.com")
+            XCTAssertEqual(details.billingAddress, expectedBillingAddress)
+            XCTAssertEqual(details.deliveryAddress, expectedDeliveryAddress)
+            
+            sut.stopLoadingIfNeeded()
+            didSubmitExpectation.fulfill()
+        }
+                
+        wait(for: .seconds(1))
+        
+        let view: UIView = sut.viewController.view
+        
+        let firstNameView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.firstNameItem"))
+        populate(textItemView: firstNameView, with: "Katrina")
+        
+        let lastNameView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.lastNameItem"))
+        populate(textItemView: lastNameView, with: "Del Mar")
+        
+        let phoneNumberView: FormPhoneNumberItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.phoneNumberItem"))
+        populate(textItemView: phoneNumberView, with: "2025550146")
+        
+        let emailView: FormTextInputItemView = try XCTUnwrap(view.findView(by: "AdyenComponents.AffirmComponent.emailItem"))
+        populate(textItemView: emailView, with: "katrina@mail.com")
+        
+        // ===================== BILLING ADDRESS =====================
```

Comment:
I'm aware of this comments. This is WIP

Reply:
I guess the comment is fine :D
`+` it is a tests